lnbits/core/models.py

```diff
@@ -253,58 +253,25 @@ async def set_pending(self, pending: bool) -> None:
 
         await update_payment_status(self.checking_id, pending)
 
-    async def check_status(
-        self,
-        conn: Optional[Connection] = None,
-    ) -> PaymentStatus:
-        if self.is_uncheckable:
-            return PaymentPendingStatus()
+    async def check_status(self) -> PaymentStatus:
 
         logger.debug(
             f"Checking {'outgoing' if self.is_out else 'incoming'} "
             f"pending payment {self.checking_id}"
         )
 
+        if self.is_uncheckable:
+            return PaymentPendingStatus()
+
         funding_source = get_funding_source()
         if self.is_out:
             status = await funding_source.get_payment_status(self.checking_id)
         else:
             status = await funding_source.get_invoice_status(self.checking_id)
 
         logger.debug(f"Status: {status}")
-
-        if self.is_in and status.pending and self.is_expired and self.expiry:
```

Comment:
good clean-up?

Reply:
yes is see no reason to keep this in the model class